src/Compilers/CSharp/Portable/Parser/Lexer.cs

```diff
@@ -933,23 +987,31 @@ private bool ScanNumericLiteral(ref TokenInfo info)
             _builder.Clear();
             bool hasUSuffix = false;
             bool hasLSuffix = false;
+            bool underscoreInWrongPlace = false;
+            bool usedUnderscore = false;
 
             ch = TextWindow.PeekChar();
-            if (ch == '0' && ((ch = TextWindow.PeekChar(1)) == 'x' || ch == 'X'))
+            if (ch == '0')
             {
-                TextWindow.AdvanceChar(2);
-                isHex = true;
+                ch = TextWindow.PeekChar(1);
+                if (ch == 'x' || ch == 'X')
+                {
+                    TextWindow.AdvanceChar(2);
+                    isHex = true;
+                }
+                else if (ch == 'b' || ch == 'B')
```

Comment:
@AdamSpeight2008 I agree that's what the code implies, I'm just confirming that's intentional in the design docs.